taglib/ape/apetag.cpp

```diff
@@ -65,13 +65,7 @@ namespace
         return false;
     }
 
-    const String upperKey = String(key).upper();
-    for(auto k : invalidKeys) {
-      if(upperKey == k)
-        return false;
-    }
-
-    return true;
+    return std::none_of(invalidKeys.begin(), invalidKeys.end(), [upperKey = String(key).upper()](auto k) { return upperKey == k; });
```

Comment:
Is the predicate `upperKey = String(key).upper()` evaluated for each iteration?

Reply:
no, only once. That's why it's in the capture list.

Reply:
btw to clarify, I need to use a lambda here because of ==. String needs to be on the left hand side.